ksops.go

```diff
@@ -13,14 +13,54 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"path/filepath"
 
+	"github.com/joho/godotenv"
 	"go.mozilla.org/sops/v3/cmd/sops/formats"
 	"go.mozilla.org/sops/v3/decrypt"
 	"sigs.k8s.io/yaml"
 )
 
+type Metadata struct {
+	Name        string            `json:"name,omitempty" yaml:"name,omitempty"`
+	Namespace   string            `json:"namespace,omitempty" yaml:"namespace,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty" yaml:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty" yaml:"annotations,omitempty"`
+}
+
+type KubernetesSecret struct {
+	ApiVersion string            `json:"apiVersion" yaml:"apiVersion"`
+	Kind       string            `json:"kind" yaml:"kind"`
+	Metadata   Metadata          `json:"metadata" yaml:"metadata"`
+	Type       string            `json:"type,omitempty" yaml:"type,omitempty"`
+	StringData map[string]string `json:"stringData" yaml:"stringData"`
+}
+
 type ksops struct {
-	Files []string `json:"files,omitempty" yaml:"files,omitempty"`
+	Files      []string `json:"files,omitempty" yaml:"files,omitempty"`
+	SecretFrom []struct {
+		Files    []string `json:"files,omitempty" yaml:"files,omitempty"`
+		Envs     []string `json:"envs,omitempty" yaml:"envs,omitempty"`
+		Metadata Metadata `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+		Type     string   `json:"type,omitempty" yaml:"type,omitempty"`
```

Comment:
Can you move this into it's own struct instead of inlining it?

Reply:
done